lib/src/atom/matcher.rs

```diff
@@ -308,22 +335,38 @@ impl Bindings {
     /// ```
     /// 
     /// TODO: Rename to `merge` when clients have adopted new API 
-    pub fn merge_v2(a: &Bindings, b: &Bindings) -> Vec<Bindings> {
-        log::trace!("Bindings::merge: a: {}, b: {}", a, b);
+    /// 
+    /// TODO: QUESTION for Vitaly: Does it make more sense to expose a method `self` api here to be more harmonious
+    /// with the rest of the public interface, vs the `a, b` arguments?  The API in the old style forces a new 
+    /// Bindings instance to be created, even if the intent of the caller is to modify / reuse an existing instance,
+    /// but the caller is always free to clone the Bindings prior to calling if they wish to continue to use the
+    /// original Bindings unchanged.
```

Comment:
Makes sense.  I agree `b::merge(&a)` is a lot nicer than `BindingsSet::from(b).merge_bindings(&a)`

As far as which function contains the implementation, my feeling is that it's better to keep it inside `Bindings::merge_v2` because that is fundamentally a one-to-many operation, as opposed to `BindingsSet::merge_bindings()` which is fundamentally a many-to-many operation.